mixer/template/authz/template.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package authz;
+
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+
+// List of subjects
+message Subject {
+  // The user.
+  string user = 1;
+  // Groups the subject belongs to.
+  repeated string groups = 2;
+  // Additional attributes about the subject.
+  map<string, istio.mixer.v1.config.descriptor.ValueType> extra = 3;
+}
+
+// List of actions
+message Action {
+  // Namespace the target action is taking place in.
+  string namespace = 1;
+  // The Service the action is being taken on.
+  string service = 2;
+  // The what action is being taken.
+  string method = 3;
+  // The resource within the service.
+  string path = 4;
+  // Additional data about the action for use in policy.
+  map<string, istio.mixer.v1.config.descriptor.ValueType> extra = 5;
+}
+
+// Authz template defines subject and action. Adapters using authz template
```

Comment:
Can we add more detail about the purpose here? This will end up in docs on the website, and while authorization is reasonably clear to sophisticated users, I think we could benefit from some lingo clarity.  Maybe something like:

> The authorization template defines parameters for performing policy enforcement within Istio. It is primarily concerned with enabling Mixer adapters to make decisions about *who* is allowed to do *what*. In this template, the *who* is defined in a *Subject* message. The *what* is defined in the *Action* message. During a Mixer Check call, these values will be populated based on configuration from request attributes and passed to individual authorization adapters to adjudicate.

You can probably describe it better, but I think that would be helpful (and maybe eliminate the need for the following paragraph).

Reply:
Replaced